etp-core/etp-backend/src/test/clj/solita/etp/service/energiatodistus_pdf_test.clj

```diff
@@ -232,3 +236,123 @@
                                                "fi"
                                                nil)
              :already-signed))))
+
+(t/deftest sign-with-system-states-test
+  (t/testing "Signing a pdf using the system instead of mpollux"
+    (let [{:keys [laatijat energiatodistukset]} (test-data-set)
+          laatija-id (-> laatijat keys sort first)
+          db (ts/db-user laatija-id)
+          ;; The second ET is 2018 version
+          id (-> energiatodistukset keys sort second)
+          whoami {:id laatija-id}
+          {:keys [versio] :as complete-energiatodistus} (complete-energiatodistus-service/find-complete-energiatodistus db id)
+          language-code (-> complete-energiatodistus :perustiedot :kieli (energiatodistus-service/language-id->codes) first)]
+      (t/testing "Testing assumptions about the energiatodistus in test"
+        (t/is (or (= language-code "sv") (= language-code "fi")))
+        (t/is (= id 2))
+        (t/is (= versio 2018)))
+      (doall [(t/testing "Signing a pdf should succeed"
```

Comment:
Onko tässä joku syy miksi testit ovat vektorissa ja doallissa eivätkä vain suoraan tässä peräkkäin?

Reply:
Siinä kävi [näin](https://github.com/solita/ara-etp/runs/24210714942), jos ei käyttänyt `doall`. Eli seuraavaa suoritettaessa edellinen ei ollut vielä valmis. En nyt tiedä onko tämä oikea tapa korjata tällainen ongelma, mutta se toimi.